src/wallet/did_wallet/did_wallet.py

```diff
@@ -0,0 +1,760 @@
+                    break
+                if coinrecord.coin.name() in unconfirmed_removals:
+                    continue
+                if coinrecord.coin in exclude:
+                    continue
+                sum_value += coinrecord.coin.amount
+                used_coins.add(coinrecord.coin)
+                self.log.info(
+                    f"Selected coin: {coinrecord.coin.name()} at height {coinrecord.confirmed_block_index}!"
+                )
+
+            # This happens when we couldn't use one of the coins because it's already used
+            # but unconfirmed, and we are waiting for the change. (unconfirmed_additions)
+            if sum_value < amount:
+                raise ValueError(
+                    "Can't make this transaction at the moment. Waiting for the change from the previous transaction."
+                )
+
+        self.log.info(f"Successfully selected coins: {used_coins}")
+        return used_coins
+
+    # This will be used in the recovery case where we don't have the parent info already
+    async def coin_added(
+        self, coin: Coin, height: int, header_hash: bytes32, removals: List[Coin]
+    ):
+        """ Notification from wallet state manager that wallet has been received. """
+        self.log.info("DID wallet has been notified that coin was added")
+
+        search_for_parent: bool = True
+        inner_puzzle = await self.inner_puzzle_for_did_puzzle(coin.puzzle_hash)
+        new_info = DIDInfo(
+            self.did_info.my_did,
+            self.did_info.backup_ids,
+            self.did_info.parent_info,
+            inner_puzzle,
+        )
+        await self.save_info(new_info)
+
+        future_parent = CCParent(
+            coin.parent_coin_info,
+            inner_puzzle.get_tree_hash(),
+            coin.amount,
+        )
+
+        await self.add_parent(coin.name(), future_parent)
+
+        for name, ccparent in self.did_info.parent_info:
+            if coin.parent_coin_info == name:
+                search_for_parent = False
+                break
+
+        if search_for_parent:
+            data: Dict[str, Any] = {
+                "data": {
+                    "action_data": {
+                        "api_name": "request_generator",
+                        "height": height,
+                        "header_hash": header_hash,
+                    }
+                }
+            }
+
+            data_str = dict_to_json_str(data)
+            await self.wallet_state_manager.create_action(
+                name="request_generator",
+                wallet_id=self.wallet_info.id,
+                type=self.wallet_info.type,
+                callback="generator_received",
+                done=False,
+                data=data_str,
+            )
+
+    # This should basically never be called as we don't want to receive ID coins from somebody else
+    async def search_for_parent_info(
+        self, block_program: Program, removals: List[Coin]
+    ) -> bool:
```

Comment:
Why does it return a bool? What does the bool indicate? Shouldn't we be returning the parent info? Calling this long complicated function for its side effects is extremely gross.